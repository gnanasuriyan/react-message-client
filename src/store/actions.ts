export { alertHide, alertShow } from './slices/alerts';
export { setAppOptions } from './slices/app';
export { getMessageList, getMessageListFailure, getMessageListSuccess, postMessage, postMessageSuccess, postMessageFailure } from './slices/message';
export { login, loginSuccess, logOut, logOutSuccess } from './slices/user';
export { signup, signupSuccess, signupFailure } from './slices/signup';
