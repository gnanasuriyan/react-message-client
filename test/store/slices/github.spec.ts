import { getMessageList, getMessageListFailure, getMessageListSuccess } from '~/actions';

import reducer from '~/store/slices/message';

describe('slices/message', () => {
  describe('actions', () => {
    it('getMessageList', () => {
      expect(getMessageList()).toMatchSnapshot();
    });
  });

  describe('reducer', () => {
    it('should return the initial state', () => {
      expect(reducer(undefined, { type: '', payload: {} })).toMatchSnapshot();
    });

    describe('getMessageList', () => {
      it('should handle REQUEST', () => {
        expect(reducer(undefined, getMessageList())).toMatchSnapshot();
      });

      it('should handle SUCCESS', () => {
        const initialState = reducer(undefined, getMessageList());

        expect(
          reducer(initialState, {
            type: getMessageList.type,
            payload: {
              data: [{ name: 'one' }],
            },
            meta: {
              query: 'react',
              updatedAt: 1234567890,
            },
          }),
        ).toMatchSnapshot();
      });

      it('should handle FAILURE', () => {
        const initialState = reducer(undefined, getMessageList('react'));

        expect(
          reducer(initialState, {
            type: getMessageList.type,
            payload: {
              message: 'Something failed',
            },
            meta: {
              query: 'react',
            },
          }),
        ).toMatchSnapshot();
      });
    });
  });
});
