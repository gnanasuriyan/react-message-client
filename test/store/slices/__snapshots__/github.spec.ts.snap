// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`slices/message > actions > getMessageList 1`] = `
{
  "payload": undefined,
  "type": "message/getMessageList",
}
`;

exports[`slices/message > reducer > getMessageList > should handle FAILURE 1`] = `
{
  "loadStatus": "running",
  "messages": [],
  "postStatus": "idle",
}
`;

exports[`slices/message > reducer > getMessageList > should handle REQUEST 1`] = `
{
  "loadStatus": "running",
  "messages": [],
  "postStatus": "idle",
}
`;

exports[`slices/message > reducer > getMessageList > should handle SUCCESS 1`] = `
{
  "loadStatus": "running",
  "messages": [],
  "postStatus": "idle",
}
`;

exports[`slices/message > reducer > should return the initial state 1`] = `
{
  "loadStatus": "idle",
  "messages": [],
  "postStatus": "idle",
}
`;
